examples/server/server.cpp

```diff
@@ -727,13 +727,21 @@ struct server_task_result_cmpl_partial : server_task_result {
 
 struct server_task_result_embd : server_task_result {
     int index = 0;
-    std::vector<float> embedding;
+    std::vector<std::vector<float>> embedding;
 
     virtual int get_index() override {
         return index;
     }
 
     virtual json to_json() override {
+        if (embedding.size() == 1){
```

Comment:
```suggestion
        if (embedding.size() == 1) {
```

Not sure if I understand correctly, but the case where the embd vector has one element is for reranking, right?

Reply:
`embedding` is now a vector of vectors. Here we check if there is only one vector in the results, not that the embedding size is 1 element. So far, all paths returned a single embedding vector. With this change, if `--pooling none` and there are more than 1 tokens in the input batch, the result will be a vector of embedding vectors.

For reranking, we output a single float which is the score. It is enabled only for `--pooling rank`. It should not be affected by this change.